### DISCUSSION TITLE
MTD Configurations

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#2563] [Configuration] [answered]
1. I am running MTD and would like to run 2 different thresholds, but cannot seem to determine how to do this. The glossary states that for FCST_MTD_CONV_RADIUS and FCST_MTD_CONV_THRESH can take a comma-separated list. So I have:
FCST_MTD_CONV_RADIUS = 5, 5
FCST_MTD_CONV_THRESH = ge0.0254, ge0.1016
I have tried multiple variations of adding [ ] around the list as well. It did not complain about [] around the CONV_RADIUS list, but nothing worked for CONV_THRESH. (The met_config uses [ ] for listing these, which is why I tried this)
2. There are 2 different config options in the glossary for the same thing, just ordered differently i.e. FCST_MTD_CONV_RADIUS and MTD_FCST_CONV_RADIUS and also appears for CONV_THRESH and OBS. Just odd so bringing this to attn (in "v5.1.0" and "latest" UG) 
3. EDIT I forgot that I already had an [issue](https://github.com/dtcenter/METplus/issues/1575) for this one. -> It seems regional masking is not an option in MTD (not in the default met_config). Can you confirm? I've had this question for awhile and figured I'd finally ask. I've just been running genVxMask first.

**Comment by CPKalb** [comment, marked as answer]
Hi Tracy:

1.  MTD only accepts a single convolution radius and threshold.  @davidalbo, is there an issue created regarding allowing MTD to process multiple convolution radii and thresholds?   Currently, the only way to run multiple thresholds is to run MTD multiple times, unfortunately.  This can be set up by using [configuration sections and instance names](https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#instance-names-in-process-list) in METplus.  If you want to do this, I can help set it up if I have access to your configuration file.
2.   At the moment, FCST_MTD_CONV_RADIUS is the correct variable and MTD_FCST_CONV_RADIUS will not work.  There has been some discussion on adding the functionality to support variables with the order MTD_FCST_CONV_RADIUS, but they are not currently supported.  We should probably remove them from the glossary so thanks for brining it to our attention. 
3.  Grid/poly masking is not yet available in MTD.  Based on the issue, it looks like it might be available in MET-12.0.0.  @davidalbo, is Grid/poly masking ready for METplus or still in progress?

Tina

**Reply by hertneky** [reply]
Hi Tina - Thanks for the reply. That all sounds good.

Will just mention that for # 1, the glossary is misleading, for example:
FCST_MTD_CONV_RADIUS
Comma separated list of convolution radius values used by mode-TD for forecast files. A corresponding variable exists for observation data called [OBS_MTD_CONV_RADIUS]
FCST_MTD_CONV_THRESH
Comma separated list of convolution threshold values used by mode-TD for forecast files. A corresponding variable exists for observation data called [OBS_MTD_CONV_THRESH]

**Reply by CPKalb** [reply]
Yes, our documentation is incorrect.  I'm assuming it was copied from the MODE versions of these variable which do tatke lists.  We need to update/fix the documentation to be clear that these need to be single numbers.

Tina

**Reply by CPKalb** [reply]
We have updated the documentation to remove MTD_FCST/OBS_CONV_RADIUS and MTD_FCST/OBS_CONV_THRESH.  It's has been merged into the develop branch.  Part of this update was also fixing the definitions to specify that they need to be a single number rather than an array.

Tina

**Comment by CPKalb** [comment]
As of May 1, 2024, grid/poly masking has not yet been started for MTD.  However, it is marked as required for the MET 12.0.0 release.

I'm going to go ahead and close this discussion.  If you have future questions, feel free to start a new discussion.

Tina
